src/views/CashlinkManage.vue

```diff
@@ -93,20 +93,33 @@ export default class CashlinkManage extends Vue {
     private readonly nativeShareAvailable: boolean = (!!window.navigator && !!window.navigator.share);
 
     private async mounted() {
-        this.retrievedCashlink = this.cashlink || null;
-        if (!this.keyguardResult && this.request.cashlinkAddress) {
-            const cashlink = await CashlinkStore.Instance.get(this.request.cashlinkAddress.toUserFriendlyAddress());
-            if (cashlink) {
-                this.retrievedCashlink = cashlink;
-                this.isTxSent = true;
-                this.isManagementRequest = true;
-            } else {
-                this.$rpc.reject(new Error(`${this.request.cashlinkAddress} is not a valid Cashlink.`));
+        let storedCashlink;
+        if (this.request.cashlinkAddress) {
+            storedCashlink = await CashlinkStore.Instance.get(this.request.cashlinkAddress.toUserFriendlyAddress());
+            if (!storedCashlink) {
+                this.$rpc.reject(new Error(`Could not find Cashlink for address ${this.request.cashlinkAddress}.`));
                 return;
             }
+        } else if (this.cashlink) {
+            storedCashlink = await CashlinkStore.Instance.get(this.cashlink.address.toUserFriendlyAddress());
```

Comment:
`storedCashlink` always contains a cashlink from the cashlink store, which can be null, even if `this.cashlink` is defined. They are thus not the same.

Reply:
So this would be expected to result in `storedCashlink === null` in case of a Keyguard signed Cashlink and `storedCashlink !== null` in the ledger case?

Reply:
Yes, exactly.